src/NuGet.Clients/PackageManagement.VisualStudio/EnvDTEProjectAdapter.cs

```diff
@@ -156,10 +159,15 @@ public bool IsLegacyCSProjPackageReferenceProject
 
                 if (!_isLegacyCSProjPackageReferenceProject.HasValue)
                 {
+                    var restoreProjectStyle = GetMSBuildProperty(AsIVsBuildPropertyStorage, RestoreProjectStyle);
+                    if (!string.IsNullOrWhiteSpace(restoreProjectStyle) && restoreProjectStyle.Equals(ProjectStylePackageReference, StringComparison.OrdinalIgnoreCase))
```

Comment:
The property check looks good, but the property alone isn't enough. The project must be a VSProject4. It looks like you could combine this with the other check below to also verify that RestoreProjectStyle is not set to PackageReference.